Conan support (rudimentary)

JonasProgrammer opened the pull request:
Hi,

sorry this took a little while, but here is finally a rudimentary conan integration.

The package aims to be compatible with both the `cmake_paths` generator, exporting the existing `ozo-config.cmake`, `ozo-targets.cmake` etc. files, as well as `cmake_find_package`, mirroring relevant definitions and dependencies from `CMakeLists.txt` in the conan components.

The reason for this double-specification is, that the central conan repository does not allow cmake config files to be exported, so this is my attempt at keeping it possible to be used in both scenarios. Potential further conan-center compatibility is also why I decided to pull in `resource_pool` as a package itself.

Please note, that this then still does not fully comply with the requirements for `conan-center`, I just tried to keep it in a way, so a migration further down the line is easy. Especially in regards to versioning and releasing versions, there is some more discussion needed, before we can try and get them included there.

JonasProgrammer committed:
Conan compatibility
- add basic conan file
- emulate cmake_find_package behaviour
- use dynamic version for now
- update resource_pool with latest  conan updates
- hotfix for yandex/ozo#281

elsid commented on conanfile.py:
```diff
@@ -0,0 +1,56 @@
+class Ozo(ConanFile):
+    name = 'ozo'
+    version = get_version()
+    license = 'MIT'
```
ozo has PostgreSQL License

JonasProgrammer replied:
My bad, that's what you get from copying and pasting the recipe instead of writing it from the ground up.

elsid reviewed (changes requested):
A separate docker image for conan looks like an optimizaiton but this creates complications for the whole setup. I don't see how `docker_conan clang release` will work and `docker clang conan` is also incorrect. But still build.sh help suggest that these commands are correct. Can't we install all conan related packages into the existing build image? Generally the idea for this image is to have it built on your machine for ozo development, update it very rarely and don't remove it. With a separate conan image to test all possible changes locally you'd need two images.

JonasProgrammer committed:
Adjustments after review
- fixed licence
- use one docker image
- allow conan to operate independent of compiler

JonasProgrammer committed:
Fix unstable package generation; closes #280
- package generated is now the same for GCC and Clang
- Warnings for clang are propagated only in build mode (e.g. when used with add_subdirectory/ExternalContent), not the installed targets
- installed targets are populated via package configuration (i.e. at consumption time)

JonasProgrammer committed:
Expose disabled warnings (#280) via conanfile

elsid commented on scripts/build.sh:
```diff
@@ -195,9 +223,9 @@ case "$1" in
         exit 0
     ;;
-    docker)
+    docker|docker_conan)
```
`docker_conan` is not used anymore.

elsid commented on scripts/build.sh:
```diff
@@ -195,9 +223,9 @@ case "$1" in
+    docker|docker_conan)
         set -x
-        launch_in_docker $2 $3 $4 $5
+        launch_in_${1} $2 $3 $4 $5
```
This change is not required anymore.

JonasProgrammer committed:
Reverted forgotten changes
